code/iaas/lb-instance/src/main/java/io/cattle/platform/lb/instance/service/impl/LoadBalancerInstanceManagerImpl.java

```diff
@@ -154,12 +173,12 @@ protected Agent getLoadBalancerAgent(LoadBalancer loadBalancer, long hostId) {
         return instances;
     }
 
-    private List<Long> populateHosts(LoadBalancer loadBalancer, Long... hostIds) {
-        List<Long> hosts = new ArrayList<Long>();
-        if (hostIds.length == 0) {
-            hosts = lbInstanceDao.getLoadBalancerHosts(loadBalancer.getId());
+    private List<LoadBalancerHostMap> populateHostMaps(LoadBalancer loadBalancer, LoadBalancerHostMap... hostMaps) {
+        List<LoadBalancerHostMap> hosts = new ArrayList<>();
```

Comment:
You are right, before hostMaps came in, this function made sense because it was translating hostids into hostmaps. Now we might not need it.